src/include/txn.i

```diff
@@ -297,10 +302,40 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 	op->type = F_ISSET(session, WT_SESSION_LOGGING_INMEM) ?
 	    WT_TXN_OP_INMEM : WT_TXN_OP_BASIC;
 #ifdef HAVE_TIMESTAMPS
-	if (__wt_txn_update_needs_timestamp(session, op))
+	if (__wt_txn_update_needs_timestamp(session, op)) {
 		__wt_timestamp_set(&upd->timestamp, &txn->commit_timestamp);
+
+		/*
+		 * Checkpoint transactions and lookaside transactions will not
+		 * be prepared.
+		 */
+		if (!WT_SESSION_IS_CHECKPOINT(session) &&
+		    ((WT_BTREE *)op->dhandle->handle)->id !=
+		    S2C(session)->cache->las_fileid) {
+			/*
+			 * Store the key, to search the update incase of
+			 * prepared transaction.
+			 */
+			if (btree->type == BTREE_ROW) {
+				WT_CLEAR(op->u.single_op.key.row_key);
+				item = &op->u.single_op.key.row_key;
+				WT_RET(__wt_cursor_get_raw_key(
+				    &cbt->iface, item));
+				WT_RET(__wt_buf_set(session,
+				    item, item->data, item->size));
```

Comment:
I can see that other places where __wt_cursor_get_raw_key is used are also calling __wt_buf_set, but I don't see why - do you know if it might be doing something to the copied memory?

Reply:
I think __wt_buf_set() is needed because when getting raw key, __wt_cursor_get_keyv() sets WT_ITEM data and size to refer to the cursors. __wt_buf_set() takes care of copying the external memory into the WT_ITEM's memory. If we do not do so, the cursor might move in the meanwhile and we end up referring to some other unintended key.